### PR TITLE
Only use FEATURE_PERFTRACING and FEATURE_EVENT_TRACE features when tu…

### LINKED ISSUE #17199: Regression in vm/corhost.cpp
#17080 Adds information to EventPipe in corhost.cpp

EventPipe is only defined when FEATURE_PERFTRACING is turned on, so the build currently breaks when PERFTRACING is off. This affects android builds (and anyone else without liblttng-ust-dev)

### PR DESCRIPTION
**Pull request by llamadonica**
@brianrob @adamsitnik 

Allows FEATURE_EVENT_TRACE, FEATURE_PERFTRACING to be turned off.

Fixes #17199.

### CODE REVIEW

File: src/inc/eventtracebase.h [hunk 1]
```diff
@@ -611,7 +611,7 @@ namespace ETW
 #else // FEATURE_EVENT_TRACE
     public:
         static VOID MethodJitting(MethodDesc *pMethodDesc, SString *namespaceOrClassName=NULL, SString *methodName=NULL, SString *methodSignature=NULL) {};
-        static VOID MethodJitted(MethodDesc *pMethodDesc, SString *namespaceOrClassName=NULL, SString *methodName=NULL, SString *methodSignature=NULL, SIZE_T pCode = 0, ReJITID rejitID = 0) {};
+        static VOID MethodJitted(MethodDesc *pMethodDesc, SString *namespaceOrClassName=NULL, SString *methodName=NULL, SString *methodSignature=NULL, SIZE_T pCode = 0, ReJITID rejitID = 0, BOOL bProfilerRejectedPrecompiledCode = FALSE, BOOL bReadyToRunRejectedPrecompiledCode = FALSE) {};
```
Comment (adamsitnik) [inline src/inc/eventtracebase.h]: @brianrob I don't understand this change. Why do we need to change the signature of `MethodJitted`?

Comment (brianrob) [reply src/inc/eventtracebase.h]: This is unrelated to the FEATURE_PERFTRACING.  It is related to #15801 which changed the signature of MethodJitted when FEATURE_EVENT_TRACE is defined, but not when it's not.

**Comment by brianrob** [comment]
@llamadonica, thanks for contributing this.  Can you please sign the CLA and then we can merge this when the CI completes.

File: src/inc/eventtracebase.h (hunk 1 above)
Comment (adamsitnik) [reply src/inc/eventtracebase.h]: @brianrob thanks!

**Pull request merged**